### PR TITLE
Switch to Java 11

### PR DESCRIPTION
**Pull request by jurgenvinju**

### CODE REVIEW

**Comment by msteindorfer** [comment]
@jurgenvinju I suggest to separate the reformatting from the content change (or even drop it from this PR). This would make the PR easier to review, and it would interfere less with other changes in flight.

**Comment by jurgenvinju** [comment]
@msteindorfer commit d9014c5 is the interesting one; next to that I bumped the version of capsule for this to 0.7.0 (so one minor version) to reflect that it is (possibly) not backward compatible with the Java 8 version.

File: pom.xml [hunk 1]
```diff
@@ -142,5 +148,15 @@
...
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-api</artifactId>
+      <version>1.6.1</version>
+    </dependency>
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-simple</artifactId>
+      <version>1.6.1</version>
+    </dependency>
```
Comment (msteindorfer) [inline pom.xml]: Why were the logging dependencies added?

File: pom.xml [hunk 2]
```diff
@@ -142,5 +148,15 @@
...
+</dependencies>
```
Comment (msteindorfer) [inline pom.xml:156]:
```suggestion
  </dependencies>
```

File: pom.xml [hunk 3]
```diff
@@ -142,5 +148,10 @@
       <version>${junitVersion}</version>
       <scope>test</scope>
     </dependency>
-  </dependencies>
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-simple</artifactId>
+      <version>1.6.1</version>
+    </dependency>
```
Comment (msteindorfer) [inline pom.xml:155]: Why were the logging dependencies added?

Comment (msteindorfer) [reply pom.xml:155]: Are those meant for testing scope?

File: pom.xml (hunk 1 above)
Comment (jurgenvinju) [reply pom.xml]: `mvn test` prints this:

```
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
```

and then google directed me to this stackoverflow question: https://stackoverflow.com/questions/7421612/slf4j-failed-to-load-class-org-slf4j-impl-staticloggerbinder which contained the answer. I then added the dependency and that fixed the issue.

File: pom.xml (hunk 3 above)
Comment (jurgenvinju) [reply pom.xml:155]: see above answer

Comment (jurgenvinju) [reply pom.xml:155]: yes; I think only for testing scope. can we limit the dependency to that scope?

File: pom.xml (hunk 1 above)
Comment (msteindorfer) [reply pom.xml]: The message appears to come from the testing plugin:

```
[INFO] --- maven-surefire-plugin:3.0.0-M5:test (default-test) @ capsule ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
```

Thus the dependency (if added) should be limited to the testing scope.

File: pom.xml [hunk 4]
```diff
@@ -142,5 +148,10 @@
       <version>${junitVersion}</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-simple</artifactId>
+      <version>1.6.1</version>
+    </dependency>
```
Comment (msteindorfer) [inline pom.xml:155]:
```suggestion
      <version>1.7.32</version>
      <scope>test</scope>
    </dependency>
```

Comment (jurgenvinju) [reply pom.xml]: o sorry. thanks! I just did the same thing. Maybe I should just wait until you fix everything :-)

Comment (msteindorfer) [reply pom.xml]: Would be great to pin the latest stable version of SLF4J (1.7.32).

Even better would be defining the version in a property (e.g., `slf4jVersion`) similar as jUnit version etcetera are defined in the `pom.xml`.

Comment (jurgenvinju) [reply pom.xml]: done

Comment (jurgenvinju) [reply pom.xml]: still running the tests

Comment (jurgenvinju) [reply pom.xml]: io.usethesource.capsule.SpecializedMapPropertiesTest takes a long time to complete. do we generate to many different parameters there or is this necessary to get good coverage?

Comment (msteindorfer) [reply pom.xml]: > io.usethesource.capsule.SpecializedMapPropertiesTest takes a long time to complete. do we generate to many different parameters there or is this necessary to get good coverage?

I’ll follow up with you offline on that subject.

**Comment by jurgenvinju** [comment]
We merged this branch manually.

**Comment by jurgenvinju** [comment]
Github says there are unmerged commits. but I don't see them in my clone. Don't know what is going on here.

**Pull request closed without merging**